src/main/java/us/ihmc/commons/thread/RepeatingTaskThread.java

```diff
@@ -416,31 +409,34 @@ private synchronized void addScheduled(long repetitions)
        *
        * @return {@code true} if interrupted.
        */
-      private synchronized boolean waitForChange()
+      private boolean waitForChange()
       {
-         try
+         synchronized (this)
          {
-            wait();
-         }
-         catch (InterruptedException e)
-         {
-            return true;
+            try
+            {
+               wait();
+            }
+            catch (InterruptedException e)
+            {
+               return true;
+            }
          }
 
          return false;
       }
 
-      private long getScheduled()
+      private synchronized long getScheduled()
```

Comment:
I think these fields being volatile should mean there's no need to synchronize here? Otherwise I would un-volatile the fields (@ds58 had some things to say about this)

Reply:
I just don't see why these need to be synchronized. It had nothing to do with the volatile.

Reply:
reverted